src/Cake.Common/Tools/NuGet/NuGetAliases.cs

```diff
@@ -1100,5 +1101,38 @@ public static void NuGetInit(this ICakeContext context, string source, string de
             var runner = new NuGetIniter(context.FileSystem, context.Environment, context.ProcessRunner, context.Tools, resolver);
             runner.Init(source, destination, settings);
         }
+
+        /// <summary>
+        /// List packages on available from source using specified settings
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="packageId">The package Id</param>
+        /// <param name="settings">The settings.</param>
+        /// <returns>List of packages with their version</returns>
+        /// <example>
+        /// <code>
+        /// var packageList = NuGetList("Cake", new NuGetListSettings {
+        ///     AllVersions = false,
+        ///     Prerelease = false
+        ///     });
+        /// foreach(var package in packageList)
+        /// {
+        ///     Information("Found package {0}, version {1}", package.Name, package.Version);
+        /// }
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("List")]
+        [CakeNamespaceImport("Cake.Common.Tools.NuGet.List")]
+        public static IEnumerable<NuGetListItem> NuGetList(this ICakeContext context, string packageId, NuGetListSettings settings)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            var resolver = new NuGetToolResolver(context.FileSystem, context.Environment, context.Tools);
+            var runner = new NuGetList(context.FileSystem, context.Environment, context.ProcessRunner, context.Tools, resolver);
+            return runner.List(packageId, settings);
+        }
     }
```

Comment:
Could we also have  `IEnumerable<NuGetListItem> NuGetList(this ICakeContext context, string packageId)` overload.